activerecord/lib/active_record/reflection.rb

```diff
@@ -618,6 +622,30 @@ def derive_join_table
         def primary_key(klass)
           klass.primary_key || raise(UnknownPrimaryKey.new(klass))
         end
+
+        # Finds inverse reflection from +klass+ with properties equal to options with provided keys.
+        def find_inverse_reflection_by_options keys
+          result = klass.reflections.find do |name, refl|
+            # We skip polymorphic and associations with :as option because we can't determine their classes.
+            !refl.polymorphic? && !refl.options.key?(:as) && refl.klass == active_record && keys.all? do |key|
```

Comment:
The name for `!refl.polymorphic? && !refl.options.key?(:as)` is 'reflections with existing class'.

I need to skip polymorphic and reflections with `:as` because `refl.klass` fails with `NameError` for them. I can actually implement it with `begin rescue NameError`. Either in this method or create a method in `AssociationReflection` that returns a class name or nil if it is not found. Do you think that method for `AssociationReflection` can be helpful in other places?